cmd/genautocomplete/genautocomplete_test.go

```diff
@@ -21,6 +21,23 @@ func TestCompletionBash(t *testing.T) {
 	assert.NotEmpty(t, string(bs))
 }
 
+func TestCompletionBashStdout(t *testing.T) {
+	originalStdout := os.Stdout
+	tempFile, err := ioutil.TempFile("", "completion_zsh")
+	assert.NoError(t, err)
+	defer func() { _ = tempFile.Close() }()
+	defer func() { _ = os.Remove(tempFile.Name()) }()
+
+	os.Stdout = tempFile
+	bashCommandDefinition.Run(bashCommandDefinition, []string{"-"})
+
+	output, err := ioutil.ReadFile(tempFile.Name())
+	assert.NoError(t, err)
+	assert.NotEmpty(t, string(output))
+
+	defer func() { os.Stdout = originalStdout }()
```

Comment:
This should go under the `os.Stdout` change - if any function panics then it will be undone which is useful when `os.Stdout` is concerned!  So


```go
	os.Stdout = tempFile
	defer func() { os.Stdout = originalStdout }()
```

Reply:
Ah, interesting stuff. I have to read more about `defer`. I was under the impression it would, in case of panic, execute all `defer` functions in order, no matter where in the code `defer` is.